hal/inc/usart_hal.h

```diff
@@ -176,6 +176,24 @@ ssize_t hal_usart_write_buffer(hal_usart_interface_t serial, const void* buffer,
 ssize_t hal_usart_read_buffer(hal_usart_interface_t serial, void* buffer, size_t size, size_t elementSize);
 ssize_t hal_usart_peak_buffer(hal_usart_interface_t serial, void* buffer, size_t size, size_t elementSize);
 
+
+// Deprecated *dynalib* APIs for backwards compatibility
+void inline __attribute__((always_inline, deprecated("Use hal_usart_init() instead"))) HAL_USART_Init(HAL_USART_Serial serial, Ring_Buffer *rx_buffer, Ring_Buffer *tx_buffer) { hal_usart_init(serial, rx_buffer, tx_buffer); }
```

Comment:
Nit: `always_inline` is probably redundant. The compiler will easily figure it out on its own.

Reply:
`inline return_type`, not `return_type inline`.